Widgets/Deployment/WidgetsKvCreation.Ps1

```diff
@@ -33,54 +33,129 @@ function Get-SentinelApplicationObjectId {
     return $sentinelAppObjectId
 }
 
-function Print-Error {
-    param ([string]$text)
-	Write-Host $text -ForegroundColor Red
+function Write-Log {
+    param ([string]$message, [switch]$isError)
+
+    $severityLevel = if ($isError) {2} else {4} 
+    $appinsightsTraceBody = @"
+{
+    "name": "Microsoft.ApplicationInsights.Event",
+    "time": "$(Get-Date -Format "yyyy-MM-ddTHH:mm:ss.fffffffZ")",
+    "iKey": "84e09db4-d430-41e8-97a6-c6a821010cc6",
+    "tags":{
+    },
+    "data": {
+       "baseType": "MessageData",
+       "baseData": {
+          "ver": 2,
+          "message": "$message",
+          "severityLevel": $severityLevel,
+          "properties": {
+                "WorkspaceId": "$WorkspaceId",
+                "SubscriptionId": "$SubscriptionId"
+           }
+        }
+    }
+}
+"@
+    try { # Send trace to Appinsights
+        Invoke-WebRequest -Uri https://dc.services.visualstudio.com/v2/track -Method POST -Body $appinsightsTraceBody | out-null -SilentlyContinue
+    } catch {}
+
+	if ($isError)
+    {
+        Write-Host $message -ForegroundColor Red
+    }
+    else
+    {
+        Write-Host $message
+    }
+}
+
+function TryRegisterProviders {

```

Comment:
what is this used for?

Reply:
The PS registration is part of the CloudShell onboarding that we tried to automate.

KV - is my idea, in case the customers didn't create any KVs yet.  WDYT?

Reply:
I still don't understand, this replace the process where the CX create a storage account?

Reply:
No.
It is from step3 here:
[https://learn.microsoft.com/en-us/azure/cloud-shell/quickstart?tabs=azurecli](https://learn.microsoft.com/en-us/azure/cloud-shell/quickstart?tabs=azurecli)